### PR TITLE
Moving duplicate dataStructures in `cli` and `app-server` into `app-commons`

### PR DESCRIPTION
**Pull request by GreyMcCarthy**
Comes from this PR
https://github.com/bitcoin-s/bitcoin-s/issues/4528

### CODE REVIEW

File: app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala [hunk 1]
```diff
@@ -20,20 +39,1537 @@ trait AppServerCliCommand extends CliCommand {
...
+  override def defaultPort: Int = 9999
+}
+
+trait AcceptDLCCliCommand extends AppServerCliCommand{
+  override def defaultPort: Int = 9999
+}
+
+trait SendCliCommand extends AppServerCliCommand{
+  override def defaultPort: Int = 9999
+}
```
Comment (benthecarman) [inline app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala]: These should all get `defaultPort` from `AppServerCliCommand` so you dont need to override here

File: app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala [hunk 2]
```diff
@@ -20,20 +40,1603 @@ trait AppServerCliCommand extends CliCommand {
   override def defaultPort: Int = 9999
 }
 
+trait Broadcastable extends CliCommand {
+  override def defaultPort: Int = 9999
+}
+trait SignDLCCliCommand extends AppServerCliCommand {}
```
Comment (Christewart) [inline app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala]: You can remove the `{}` on these lines

File: app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala [hunk 3]
```diff
@@ -20,20 +40,1603 @@ trait AppServerCliCommand extends CliCommand {
...
+    Try(FinalizePSBT(jsToPSBT(jsArr.arr.head)))
+  }
+}
+
+case class ExtractFromPSBT(psbt: PSBT)
+    extends CliCommand
+    with AppServerCliCommand
+
+object ExtractFromPSBT
+    extends CliCommand
```
Comment (Christewart) [inline app-commons/src/main/scala/org/bitcoins/commons/rpc/Command.scala]: All objects are extending too much, for instance, this is what this was `extending` before this PR

https://github.com/bitcoin-s/bitcoin-s/blob/e90e372e54cf40f846cad3f4e6fc1932d4e765d9/app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala#L469

Note, there is a difference between `case class` and `object`. 

The only objects in this file need to `extends ServerJsonModels`, that gives us access to methods like 

- `jsToPSBT`
- `jsToTx`
- `nullToOpt` 

etc

**Review by Christewart — CHANGES_REQUESTED** [review]

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 4]
```diff
@@ -2425,41 +2414,23 @@ object CliCommand {
...
-
-  object CreateContractInfo {
-
-    lazy val empty: CreateContractInfo = {
-      CreateContractInfo(announcementTLV = OracleAnnouncementV0TLV.dummy,
-                         totalCollateral = Satoshis.zero,
-                         contractDescriptor = ujson.Null)
-    }
-  }
+  /**  case class CreateContractInfo(
```
Comment (Christewart) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: Can be removed?

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 5]
```diff
@@ -2359,9 +2378,12 @@ object Config {
 
 object CliCommand {
 
-  trait Broadcastable {
-    def noBroadcast: Boolean
-  }
+  /**  trait Broadcastable {
```
Comment (Christewart) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: Can be removed?

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 6]
```diff
@@ -2469,79 +2440,14 @@ object CliCommand {
 
   case class RemoveDLCOffer(offerHash: Sha256Digest) extends AppServerCliCommand
 
-  sealed trait SendCliCommand extends AppServerCliCommand {
-    def destination: BitcoinAddress
-  }
+  /**  sealed trait SendCliCommand extends AppServerCliCommand {
```
Comment (Christewart) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: Can be removed?

File: app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala [hunk 7]
```diff
@@ -1,1102 +1,104 @@
...
-import scodec.bits.ByteVector
 import ujson._
 
 import java.io.File
 import java.net.{InetSocketAddress, URI}
 import java.nio.file.Path
 import scala.util._
 
-case class GetNewAddress(labelOpt: Option[AddressLabelTag])
+/** trait Broadcastable {
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/ServerJsonModels.scala]: Can be removed?

**Review by Christewart — CHANGES_REQUESTED** [review]

**Pull request merged**